### PR TITLE
Create wallet_provider on Publishers 

### LINKED ISSUE #2842: Create the current wallet_provider option on Publisher
We should allow users to only have one selected crypto payment option at the time of payout. 

Edge cases:
- When a publisher changes their default currency between the 1st- 8th.  (Perhaps we can include the payout type in the payout tracker)
- Paypal is considered a crypto payment.

### PR DESCRIPTION
**Pull request by corymcdonald**
## Create wallet_provider on Publishers 

Closes #2842

#### Features

✨ Creates a `wallet_provider_id` and `wallet_provider_kind` entry on the Publishers table.
✨ Now the code can call `publisher.wallet_provider` and get either a `GeminiConnection`, `UpholdConnection`, or `PaypalConnection`

#### How To Test

1. Connect Gemini
2. In the `rails console` type `Publisher.first.wallet_provider`
3. See the `GeminiConnection` is returned.

### CODE REVIEW

File: app/models/uphold_connection.rb [hunk 1]
```diff
@@ -95,6 +95,7 @@ def disconnect_uphold
       default_currency_confirmed_at: nil,
       default_currency: nil,
     )
+    publisher.update(wallet_provider: nil)
```
Comment (yachtcaptain23) [inline app/models/uphold_connection.rb:98]: I think the `wallet_provider` should only be cleared if the uphold_connection is the selected wallet_provider

File: db/migrate/20200826231131_add_wallet_provider_to_publishers.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddWalletProviderToPublishers < ActiveRecord::Migration[6.0]
+  def change
+    add_reference :publishers, :wallet_provider, polymorphic: true, type: :uuid
```
Comment (yachtcaptain23) [inline db/migrate/20200826231131_add_wallet_provider_to_publishers.rb:3]: Nit picking: `selected_wallet_provider` or `preferred_wallet_provider` over `wallet_provider`

Comment (corymcdonald) [reply db/migrate/20200826231131_add_wallet_provider_to_publishers.rb]: https://github.com/brave-intl/publishers/pull/2857/commits/5243340640e464a6d87ed6a8aec1277e870721a5

File: app/models/uphold_connection.rb (hunk 1 above)
Comment (corymcdonald) [reply app/models/uphold_connection.rb]: https://github.com/brave-intl/publishers/pull/2857/commits/5243340640e464a6d87ed6a8aec1277e870721a5

File: app/controllers/payment/connection/currencies_controller.rb [hunk 3]
```diff
@@ -6,16 +6,16 @@ class CurrenciesController < ApplicationController
 
       def show
         render json: {
-          supported_currencies: current_publisher.wallet_provider.supported_currencies,
-          default_currency: current_publisher.wallet_provider.default_currency,
+          supported_currencies: current_publisher.selected_wallet_provider.supported_currencies,
```
Comment (yachtcaptain23) [inline app/controllers/payment/connection/currencies_controller.rb:9]: Will this crash if the selected_wallet_provider is nil?

Comment (corymcdonald) [reply app/controllers/payment/connection/currencies_controller.rb:9]: Yes, but that's intended with the changes to the default currency workflow. 👍

**Comment by evq** [comment]
++

**Pull request merged**